core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java

```diff
@@ -447,10 +447,16 @@ public boolean openUfsBlock(long sessionId, long blockId, OpenUfsBlockOptions op
 
   @Override
   public void closeUfsBlock(long sessionId, long blockId)
-      throws BlockAlreadyExistsException, BlockDoesNotExistException, InvalidWorkerStateException,
-      IOException, WorkerOutOfSpaceException {
-    if (mUnderFileSystemBlockStore.cleanup(sessionId, blockId)) {
-      commitBlock(sessionId, blockId);
+      throws BlockAlreadyExistsException, InvalidWorkerStateException, IOException,
+      WorkerOutOfSpaceException {
+    mUnderFileSystemBlockStore.closeReaderOrWriter(sessionId, blockId);
+    if (mBlockStore.getTempBlockMeta(sessionId, blockId) != null) {
```

Comment:
btw, i did not change commitBlock to be (commitBlockIfExists()->bool) because that requires lots of small changes (scattered in several files). If you guys prefer, i can do that in another PR so that this PR won't be delayed because of that.